options.go

```diff
@@ -55,6 +55,32 @@ type options struct {
 	gCHeapOpts *typeOption
 	cpuOpts    *typeOption
 	threadOpts *typeOption
+
+	pReportOpts *ReporterOptions
+}
+
+// rptEvent stands of the args of report event
+type rptEvent struct {
+	PType   string
+	Buf     []byte
+	Reason  string
+	EventID string
+}
+
+type ReporterOptions struct {
+	reporter ProfileReporter
+	active   int32 // switch
+	eventsCh chan rptEvent
+	cancelCh chan struct{}
```

Comment:
it's not a good idea to put them under Options, under Holmes should be better.

Reply:
them means `pReportOpts`? We need to use the Lock `L` inside options.

 It's ok that just move `eventsCh` and `cancelCh` to Holmes.

Reply:
just eventsCh and cancelCh.